presto-main/src/main/java/io/prestosql/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -482,9 +482,22 @@ public PlanWithProperties visitTableWriter(TableWriterNode node, StreamPreferred
         {
             StreamPreferredProperties requiredProperties;
             StreamPreferredProperties preferredProperties;
-            if (getTaskWriterCount(session) > 1 && !node.getPartitioningScheme().isPresent()) {
-                requiredProperties = fixedParallelism();
-                preferredProperties = fixedParallelism();
+            // TODO: add support for arbitrary partitioning in local exchanges
```

Comment:
@dain mentioned there was some PR to add support for arbitrary partitioning in local exchanges.